simulations/vip-converter/vip-converter-mainnet/simulations.ts

```diff
@@ -0,0 +1,120 @@
+];
+
+const PROTOCOL_SHARE_RESERVE_PROXY = "0xCa01D5A9A248a830E9D93231e791B1afFed7c446";
+
+forking(35781703, () => {
+  const provider = ethers.provider;
+  let impersonatedTimelock: Signer;
+  let proxyAdmin: Contract;
+  let protocolShareReserve: Contract;
+  let beacon: Contract;
+  let liquidator: Contract;
+
+  before(async () => {
+    proxyAdmin = new ethers.Contract(PROXY_ADMIN, DEFAULT_PROXY_ADMIN_ABI, provider);
+    beacon = new ethers.Contract(SINGLE_TOKEN_CONVERTER_BEACON, BEACON_ABI, provider);
+    impersonatedTimelock = await initMainnetUser(NORMAL_TIMELOCK, ethers.utils.parseEther("1"));
+    liquidator = new ethers.Contract(LIQUIDATOR, LIQUIDATOR_ABI, provider);
+
+    protocolShareReserve = new ethers.Contract(PROTOCOL_SHARE_RESERVE_PROXY, PROTOCOL_SHARE_RESERVE_ABI, provider);
+  });
+
+  testVip("VIP-Converter", vipConverter(), {
+    callbackAfterExecution: async (txResponse: any) => {
+      await expectEvents(txResponse, [TRANSPARENT_PROXY_ABI], ["Upgraded"], [2]);
+    },
+  });
+
+  describe("Post-VIP behavior", () => {
+    it("release funds should execute successfully", async () => {
+      await protocolShareReserve.connect(impersonatedTimelock).releaseFunds(COMPTROLLER, ASSETS);
+    });
+
+    it("RiskFundConverter and SingleTokenConverter should have new implementation", async () => {
+      expect(await proxyAdmin.getProxyImplementation(RISK_FUND_CONVERTER_PROXY)).to.equal(NEW_RISK_FUND_CONVERTER_IMP);
+      expect(await beacon.implementation()).to.equal(NEW_SINGLE_TOKEN_CONVERTER_IMP);
+    });
+
+    it("amount out and amount in tests", async () => {
+      const convertersTokenOutArray = [
+        RiskFundConverterTokenOuts,
+        USDTPrimeConverterTokenOuts,
+        USDCPrimeConverterTokenOuts,
+        BTCBPrimeConverterTokenOuts,
+        ETHPrimeConverterTokenOuts,
+        XVSVaultConverterTokenOuts,
+      ];
+
+      for (let i = 0; i < converters.length; i++) {
+        const converterAddress = converters[i];
+        const tokenOuts = convertersTokenOutArray[i];
+
+        for (const tokenAddress of tokenOuts) {
+          const converter: Contract = new ethers.Contract(converterAddress, SINGLE_TOKEN_CONVERTER_ABI, provider);
+          const configuration = await converter.conversionConfigurations(BaseAssets[i], tokenAddress);
+          if (configuration.conversionAccess == 2) continue;
+
+          const balance = await converter.balanceOf(tokenAddress);
+          if (balance > 0) {
+            const [, amountInMantissa] = await converter.getAmountIn(balance, BaseAssets[i], tokenAddress);
+            const [, amountOutMantissa] = await converter.getAmountOut(amountInMantissa, BaseAssets[i], tokenAddress);
+
+            expect(amountOutMantissa).to.be.lessThanOrEqual(balance);
+          }
+        }
+      }
+    });
+
+    it("updates protocolShareReserveAddress", async () => {
+      expect(PSR).to.equal(await liquidator.protocolShareReserve());
+    });
```

Comment:
I would add a liquidation to the simulation because the Liquidator will try to send underlying tokens to the PSR. So, I would:
1) simulate a liquidation
2) check that 5% of the repaid amount is sent to the PSR, as underlying tokens
3) release the funds received and confirm the destination target receive the expected values (liquidation income follows the `ADDITIONAL_REVENUE` schema mentioned [here](https://docs-v4.venus.io/whats-new/automatic-income-allocation#key-aspects-of-automatic-income-allocation))

For the liquidation, you could use these values (after approving the spender):
* liquidator (sender): 0x1934057d1de58cf65fb59277a91f26ac9f8a4282
* vToken to repay (vETH): 0xf508fcd89b8bd15579dc79a6827cb4686a3592c8
* borrower: 0x489a8756c18c0b8b24ec2a2b9ff3d4d447f79bec
* repayAmount: 12600000000000000
* vToken to seize (vBNB): 0xa07c5b74c9b40447a954e1466938b865b6bbea36

Reply:
The PSR does not receive 5% of repaid amount instead a settlement of seizedAmount is done using [_splitLiquidationIncentive](https://github.com/VenusProtocol/venus-protocol/blob/VEN-2375/contracts/Liquidator/Liquidator.sol#L439) function which is about **(5 / 1.1) ~ 4.54%**